app/src/main/java/com/ssimagepicker/app/MainActivity.kt

```diff
@@ -1,94 +1,140 @@
-            )
-
-            //To customize bottomsheet style
-            setBottomSheetBackgroundStyle(R.drawable.drawable_bottom_sheet_dialog)
+    private fun checkForImagePicker(pickerType: PickerType) {
+        if (isDataValid()) {
+            openImagePicker(pickerType)
         }
     }
 
-    override fun onRequestPermissionsResult(
-        requestCode: Int,
-        permissions: Array<out String>,
-        grantResults: IntArray
-    ) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    /**
+     * Check whether the max pick count and max size is valid or not.
+     * If all data is valid it will return true.
+     */
+    private fun isDataValid(): Boolean {
+        val countValue = binding.pickCountTie.text.toString().toIntOrNull()
+        val sizeValue = binding.pickSizeTie.text.toString().toIntOrNull()
+        if (binding.multiSelectionSwitch.isChecked && (countValue == null || countValue < 1)) {
+            Toast.makeText(this, R.string.error_count_value, Toast.LENGTH_LONG).show()
+            return false
+        } else if (sizeValue == null || sizeValue < 1) {
+            Toast.makeText(this, R.string.error_size_value, Toast.LENGTH_LONG).show()
+            return false
+        }
+        return true
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        imagePicker.onActivityResult(requestCode, resultCode, data)
+    /**
+     * Open the image picker according to picker type and the ui options.
+     * The new system picker is only available for Android 13+.
+     */
+    private fun openImagePicker(pickerType: PickerType) {
+        val countValue = binding.pickCountTie.text.toString().toInt()
+        val sizeValue = binding.pickSizeTie.text.toString().toInt()
+        val pickExtension = when (binding.extensionTypeGroup.checkedButtonId) {
+            R.id.all_button -> PickExtension.ALL
+            R.id.png_button -> PickExtension.PNG
+            R.id.jpeg_button -> PickExtension.JPEG
+            R.id.webp_button -> PickExtension.WEBP
+            else -> PickExtension.ALL
+        }
+        imagePicker
+            .title("My Picker")
+            .multipleSelection(binding.multiSelectionSwitch.isChecked, countValue)
+            .showCountInToolBar(binding.countToolbarSwitch.isChecked)
+            .showFolder(binding.folderSwitch.isChecked)
+            .cameraIcon(binding.cameraInGallery.isChecked)
+            .doneStyle(binding.doneSwitch.isChecked)
+            .allowCropping(binding.openCropSwitch.isChecked)
+            .compressImage(binding.compressImageSwitch.isChecked)
+            .maxImageSize(sizeValue)
+            .extension(pickExtension)
+        if (isAtLeast13()) {
+            imagePicker.systemPicker(binding.systemPickerSwitch.isChecked)
```

Comment:
I think the New Photo Picker API supports Android 11 or higher as well. Please check
https://developer.android.com/training/data-storage/shared/photopicker